### DISCUSSION TITLE
Welcome to Artifact-Travelogues- Discussions!

### DISCUSSION THREAD

**Discussion by alineestkyr** [alineestkyr/Artifact-Travelogues-#1] [Tech-related]
## 👋 Welcome!
 Welcome to the Artifact Travelogues Tech-related Discussion Thread! Here is where we'll be able to discuss any technical issues, ideas, or areas for improvement! 
 
-->

**Comment by CodyCarmichaelMPH** [comment]
Hey all! Came on from reddit and I have a few suggestions if you'd like. 

Looking at the HTML output, it looks like the following information is needed: 
Name of Object
Image of Object
then details which include: 
Duration of Ownership
Person(s)/Institutions claiming ownership during this time. 
Other details. 

If I can suggest, there might be an easier way to scrape data for later analysis, formatting, and outputting. I'm used to doing the following in R but Python would be able to create an interactive front end for users if this wants to be a full software suite that can link up elsewhere. Anyways, here's my idea: 

Each Item/Artifact is given a [data frame](https://www.w3schools.com/datascience/ds_python_dataframe.asp). This data frame (similar to an excel sheet) is given the following columns: 
- A Metadata (data about the data) tag that has the object name or other pertinent information, making it easier to find and work with. 

- Name of Object

- Image/Media link or file

- Ownership Duration

- Owner(s) during Duration

- Location of Owner(s) (ideally GPS coordinates but we could work with other data) 

- Description- Updates (say if an item was damaged or modified at this location) 

- Origin

From here, we could autofill HTML pages fairly easily, but also be able to create maps, and timelines, and even see common travel/trade patterns across artifacts. This could also pave the way for a GUI front end where people could upload information in a standardized way so that way less time is used cleaning data.

How does that sound so far? Thoughts, concerns, anything I might have missed the boat on?

**Reply by alineestkyr** [reply]
hey! this all sounds great. What would need to be done or changed here on this repository in order to give each item a data frame, as you suggested? I'm not at all familiar with data science. Would each folder just have a new file for the code written in Python?

**Reply by CodyCarmichaelMPH** [reply]
An excellent question! 

So, to do this, we'd need to have a file system (which could be done via GUI or a central CSV file) with these columns that could be filled in by users. Obviously the former if you want to pop this out into a general app or website, but as a prototype, we can use the latter. I'll start working on a prototype and put it under a folder called "Data Science Methods".  We could always do an additional file per log, but then we start running into redundancy of inputs/ hassle on the end user(s).

**Comment by RaoulHofmann** [comment]
Hi all,

I'm not familiar with python outside of some ML stuff but the general idea outlined seems pretty solid. Was your idea to store the data in the data frames manually? In my opinion this data is just screaming out for a database.

When I first saw the layout of the project my first idea was to through up a simple website using something like [strapi](https://strapi.io/) which allows users not familiar with code to easily setup endpoints which would then save the data into a MongoDB database.

The advantage with storing it in something would be that each Item/Artifact would be its own document which could contain all the information from the data frame and is very flexible with what data it can take in.

Theoretically all of this could be hosted on both a free tier MongoDB and free tier AWS instances.

**Reply by alineestkyr** [reply]
hi! I like the idea of using something that is accessible to users who are unfamiliar or limited in their knowledge of code (I myself included). I think it's a great way to get people who want to contribute to open source history projects, but are intimidated by the technical aspects of it.

**Reply by alineestkyr** [reply]
could you tell me a bit more about the process of transferring the repo to something like strapi? I've never used it before.

**Reply by RaoulHofmann** [reply]
You wouldn't need to change much with the current repo. We would basically need to define how we would want to store the data, then setup strapi with that structure. The repo could become a simple html page with some javascript which gets the data from strapi and then displays it in that page.

You could then either setup users within strapi to add new artifacts or extend the repo to allow users to register and add data themselves, which would be another couple of html pages with some javascript.

I spun up a quick strapi instance with some screenshots to show.

I'll fork the repo when I get home from work tonight (I'm in Western Australia) and play around with it

![image](https://user-images.githubusercontent.com/7591134/170613106-64ef72a7-1da1-4347-a813-c4833268fd2c.png)

![image](https://user-images.githubusercontent.com/7591134/170613357-23a4d762-b43b-4f0a-90d1-fc4cb5af0aa7.png)

![image](https://user-images.githubusercontent.com/7591134/170613461-fdf6ef25-5a27-408d-a502-4a53cfedc8bb.png)

![image](https://user-images.githubusercontent.com/7591134/170613540-00d35b23-798e-40e2-bbf2-1d0427b38a38.png)

![image](https://user-images.githubusercontent.com/7591134/170613579-3e7c39a3-578e-4b42-bf64-7366b6e9d6dc.png)

**Reply by RaoulHofmann** [reply]
There are many different solutions out there similar to strapi, it just the first one that came to mind.

Over time you/we could really improve the current page structure and make it look super fancy with functionalities which really utilise the data such as maps.

**Reply by CodyCarmichaelMPH** [reply]
This is an excellent suggestion. If you can work on the data acquisition side, I can absolutely work the backend/analytics side a bit in my fork. Or we can both rock out whatever we see fit and take the better method, but this is really impressive!

**Reply by RaoulHofmann** [reply]
Sounds good, I'll through something together later on today and we can test it out and see how it works for you guys

**Comment by RaoulHofmann** [comment]
[I've setup a strapi instance](https://artifact-travelogues.668558.xyz/) shoot me a message if you want an invite to play around with it

**Reply by alineestkyr** [reply]
Thanks!  I'd like to take a look

**Reply by RaoulHofmann** [reply]
I'll dm you the invite link in reddit

**Comment by alineestkyr** [comment]
quick question: if anybody updates the contents of the respository (ie. adds a new artifact folder or file), will it also update on strapi? Or does that have to be done separately on strapi?

**Reply by RaoulHofmann** [reply]
If you wanted to store the data in a database be that using strapi or something else you would have to rethink how you handle content updates and additions. The repo could shift into becoming a website that gets the data from the database and displays it, is searchable, can be updated and indexed, etc.

The problem I see with the current way of creating new content is that it could become very big and unwieldly very quickly and storing this sort of data is what databases excel at. But I fully understand if you wanted to keep it the way it is and are using it as a way of learning and using html and css.

Some of the ways you could update the data would be:
- Directly through strapi
- By updating the repo and using some javascript to send information put into a form (shown through gihub pages) to strapi
- By continuing to use the current method of updating the repo with new folders you could have GitHub Actions run a script which gathers data from the folders and updates the database

**Comment by alineestkyr** [comment]
hey @RaoulHofmann a small issue I'm running into with strapi, the years only go as far back as 1822. is there a way for me to adjust it?

**Reply by RaoulHofmann** [reply]
Looks like its a strapi bug. I'll remove the field and change it to a text field.

**Reply by alineestkyr** [reply]
thank you

**Comment by alineestkyr** [comment]
hey @RaoulHofmann, there seems to be something wrong with the strapi. I can't log in as an admin, nor do the database entries load on the github pages. Any ideas?

**Reply by RaoulHofmann** [reply]
Let me check, server might have gone down

**Reply by RaoulHofmann** [reply]
Try again, I messed something up over the weekend.

**Reply by alineestkyr** [reply]
works now, thanks

**Comment by RaoulHofmann** [comment]
@alineestkyr  Do you have a general idea of what you want the entries overview page to look like? I'm gonna try and do some work on the layout and a few bits and pieces this weekend.

**Reply by alineestkyr** [reply]
the index.html or the strapi directory page? I think its good with how it is right now. 

I was was thinking that we should probably add a new collection field called "sources" at the very end of the each entry page, just to declutter how it looks.

**Reply by RaoulHofmann** [reply]
Sorry, I mean the page where the artifacts get shown.

Would sources be a part of the main artifact or for each owner?

**Reply by alineestkyr** [reply]
The list of all the artifacts or the individual page? Sorry I'm a bit confused about the wording haha. 

And the sources I think would be better shown at the end of the page for each individual artifact. I was initially going to use the method of listing the sources at the end of each owner, but I realized it might clutter the page. So that means we need to add another content category on strapi specifically for "Sources". I'm going to use a footnote [1] method within the description under the different owners and put the corresponding footnotes at the end of the page. I hope that makes sense ?

**Reply by RaoulHofmann** [reply]
My bad, I meant the individual page but if you wanted me to change the list of artifacts I can do that to.

That makes sense, i've added sources to the end in strapi.
I split it up between the identifier and the actual source, let me know what you think of that I can change it pretty quickly.

![image](https://user-images.githubusercontent.com/7591134/181660966-a54f5664-d901-493c-bd4d-1dafcdae0cc0.png)

![image](https://user-images.githubusercontent.com/7591134/181661083-5acad7a7-410e-41dd-867f-41dd33d9ea65.png)

**Reply by alineestkyr** [reply]
ah right. I tried entering it in for the first and second cited source. So with what you added will it look like:

**Napoleon's soldiers were required to obtain any artifacts of cultural significance. Pierre-François Bouchard took the artifact to Cairo. [1]**

and then at the end of the page: 

**Sources:**

**[1] chicagostylecitation link: https://www.britannica.com/topic/Rosetta-Stone**


on another note, I don't see anything wrong with the directory. The only thing I'm going to play around with is maybe a more readable font for each individual page.

**Reply by alineestkyr** [reply]
ooo another thing i forgot for the individual artifact page, something to credit the photo. that can just be done as a caption under the image.

**Reply by RaoulHofmann** [reply]
Sorry works been a bit hectic lately.

I'm currently updating the server so the page might be down for a bit.

I'll add those changes in once its done.

You should also have the ability to import and export data once this update is done
